### PR TITLE
Custom overrides

### PR DESCRIPTION
**Pull request by KATT**
Closes #

## 🎯 Changes

What changes are made in this PR? Is it a feature or a bug fix?

## ✅ Checklist

- [ ] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [ ] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: packages/react/src/shared/types.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * @internal
+ */
+export interface UseMutationOverride {
+  onSuccess: (opts: {
+    /**
+     * Calls the original function that was defined in the query's `onSuccess` option
+     */
+    originalFn: () => MaybePromise<unknown>;
+    queryClient: QueryClient;
+  }) => MaybePromise<unknown>;
```
Comment (juliusmarminge) [inline packages/react/src/shared/types.ts:14]: allow overriding `onError`, `onMutate` etc too?

File: packages/react/src/shared/types.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+/**
+ * @internal
+ */
+export interface CreateTRPCReactOptions<_TRouter extends AnyRouter> {
+  /**
+   * Override behaviors of the built-in hooks
+   */
+  unstable_overrides?: {
+    useMutation?: Partial<UseMutationOverride>;
+  };
```
Comment (juliusmarminge) [inline packages/react/src/shared/types.ts:26]: allow overriding of other hooks too?

Comment (juliusmarminge) [reply packages/react/src/shared/types.ts:26]: not sure what the use case of overriding useQuery might be though.

File: packages/react/src/shared/types.ts (hunk 1 above)
Comment (KATT) [reply packages/react/src/shared/types.ts:14]: Meh. I don't use it

File: packages/react/src/shared/types.ts (hunk 2 above)
Comment (KATT) [reply packages/react/src/shared/types.ts:26]: Meh. Maybe later! :D

**Review by sachinraja — APPROVED** [review]

**Pull request merged**